tests/Doctrine/Tests/DBAL/Driver/OCI8/OCI8StatementTest.php

```diff
@@ -74,6 +74,39 @@ public function testExecute(array $params) : void
         $statement->execute($params);
     }
 
+    /**
+     * Testing binding named and positioned parameters
+     *
+     * @param mixed[] $params
+     *
+     * @dataProvider bindParamsDataProvider
+     */
+    public function testBindParam(array $params, string $query) : void
+    {
+        $conn = $this->createMock(OCI8Connection::class);
+        /** @var OCI8Statement|MockObject $statement */
+        $statement = $this->getMockBuilder(OCI8Statement::class)
+            ->onlyMethods([  'prepareQuery', 'bindParamByName'])
```

Comment:
Please take a look at `NamedParametersTest`? I'd rather have a functional test when it comes to testing the logic that involves a database (binding named parameters in this case). Is the existing suite missing something?

Reply:
A set of "NamedParametersTest" is sufficient. But in " NamedParametersTest "the higher-level method"executeQuery" is tested. This method works well, because inside it named placeholders ( :param) are replaced by positioned (? ). And only then does this modified request arrive at the oracle statement input.  Added test in Functional\Driver\OCI8\StatementTest as these fixes concern OCI8StatementTest class.